### PR TITLE
remove the ability to navigate back into the onboarding flow after completing onboarding

### LINKED ISSUE #16890: [Bug]: Onboarding V2 Going backwards to previous screen displays 1 word seedphrase placeholder
### Describe the bug

**Problem**: after completing the Create New Wallet onboarding V2, if I go back using the browser function I can see that the Backup your Seedphrase page displays 1 empty placeholder with.

Not sure if we should be able to go back.

https://user-images.githubusercontent.com/54408225/206736619-6f631afd-e96f-4eb8-8ed5-e94086a01be2.mp4

### Steps to reproduce

1. Load MM with Onbaording V2
2. Go through Create Wallet flow until Home page
3. Use the browser arrow for going back to previous pages
4. Click on Backup Seedphrase -- see 1 empty placeholder is displayed

### Error messages or log output

_No response_

### Version

10.24.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by adonesky1**
* Fixes #16890

Fixes issue where the secret recovery phrase state is not available when navigating back after completing onboarding. We now prevent users from navigating back into the onboarding flow at all after its completion. The one exception is when you click the `Backup Now` button on the `Backup your Secret Recovery Phrase to keep your wallet and funds secure.` reminder if you opted to not reveal and back up the security phrase in the initial onboarding flow.

### Before

https://user-images.githubusercontent.com/34557516/207152480-55462427-f183-40c4-8b51-c9eafb780a84.mp4

and backup reminder button takes us to V1 secure seedphrase flow instead of V2 even when feature flag is on:
https://user-images.githubusercontent.com/34557516/207154103-5e38e796-c18c-4ece-bf9d-a7b4c56f7ff6.mov

### After

See you cannot navigate back after completing onboarding but you are able to navigate back to secure your wallet if you did not do so initially (the reminder here is hardcoded to show up immediately for demonstration purposes).
https://user-images.githubusercontent.com/34557516/207164461-1e6ea62d-6e5e-4538-8b9e-3f794188bc62.mov

## Manual Testing Steps

1. Load MM with Onboarding V2
2. Go through Create Wallet flow until Home page
3. Attempt to click back, see that you cannot navigate back

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes

### CODE REVIEW

**Comment by legobeat** [comment]
Shouldn't the user be able to go back for e.g. enabling/disabling MetaMetrics in case they click the wrong button there or want to go back/change their mind before completion?

File: ui/pages/onboarding-flow/onboarding-flow.js [hunk 1]
```diff
@@ -49,19 +50,31 @@ import MetaMetricsComponent from './metametrics/metametrics';
...
   const seedPhraseBackedUp = useSelector(getSeedPhraseBackedUp);
   const nextRoute = useSelector(getFirstTimeFlowTypeRoute);
-
+  const isFromReminder = new URLSearchParams(search).get('isFromReminder');
   useEffect(() => {
-    if (completedOnboarding && seedPhraseBackedUp) {
+    if (completedOnboarding && !isFromReminder) {
       history.push(DEFAULT_ROUTE);
     }
   }, [history, completedOnboarding, seedPhraseBackedUp]);
```
Comment (digiwand) [inline ui/pages/onboarding-flow/onboarding-flow.js:64]: I believe we can update the dependencies here as well with the new changes

**Comment by adonesky1** [comment]
> Shouldn't the user be able to go back for e.g. enabling/disabling MetaMetrics in case they click the wrong button there or want to go back/change their mind before completion?

All of the preferences we allow users to provide during onboarding can be found in settings if users wish to modify them:

Screen Shot 2022-12-13 at 9 25 36 AM

File: ui/pages/onboarding-flow/onboarding-flow.js (hunk 1 above)
Comment (adonesky1) [reply ui/pages/onboarding-flow/onboarding-flow.js]: done [here](https://github.com/MetaMask/metamask-extension/pull/16916/commits/134f878548f688b439f5da53f5a959d9ad499e24)

**Review by digiwand — APPROVED** [review]
LGTM!

**Review by jpuri — APPROVED** [review]

**Pull request merged**